### ISSUE TITLE
Forbid local variables that are only used in `return` statements

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#471]
# Rule request

## Thesis

Sometimes I see code like this:

```python
def function():
     ...
     some_value = ...
     return some_value
```

What the point in creating this intermediate variable? Why do we need it?

## Reasoning

This is done for consistency and more readable source code.

**Comment by loucadufault** [comment]
Can I start working on this ? To clarify the fix would be to remove all these intermediate variables in all function returns?

**Comment by sobolevn** [comment]
@loucadufault you are welcome! 

The fix would be to return a direct value. 

Wrong:

```python
def some_function():
     some_value = 1
     return some_value
```

Correct:

```python
def some_function():
     return 1
```

**Comment by Chinmaygoyal** [comment]
Can I work on this issue?

**Comment by SizovIgor** [comment]
@Chinmaygoyal I am on finish line to complete this issue

**Comment by sobolevn** [comment]
@Chinmaygoyal feel free to take #545

**Comment by DmytroLitvinov** [comment]
Hello @sobolevn 
The immediate value is good in the next case:
```python
def function(a):
     result = None
     if a:
         ...  # some actions
        result = 123

     return result
```
As for me, it is also understandable what from naming of value what we will return after calling of function.

**Comment by sobolevn** [comment]
@DmytroLitvinov yes, this code is valid.

**Comment by DmytroLitvinov** [comment]
https://github.com/sbdchd/flake8-pie
